modules/phase_field/src/materials/KKSPhaseConcentrationMaterial.C

```diff
@@ -46,17 +48,37 @@ KKSPhaseConcentrationMaterial::KKSPhaseConcentrationMaterial(const InputParamete
     _prop_ci(_num_c * 2),
     _ci_old(_num_c * 2),
     _ci_IC(getParam<std::vector<Real>>("ci_IC")),
+
     _f1(getMaterial("F1_material")),
     _f2(getMaterial("F2_material")),
+    _prop_fi(2),
+    _fi_copy(2),
+    _fi_copy_name(getParam<std::vector<MaterialPropertyName>>("Fi_copy")),
+
     _Fi_names(getParam<std::vector<MaterialPropertyName>>("Fi_names")),
     _first_dFi(2),
     _second_dFi(2),
     _iter(declareProperty<Real>("nested_iterations")),
     _abs_tol(getParam<Real>("absolute_tolerance")),
     _rel_tol(getParam<Real>("relative_tolerance")),
-    _nested_solve(NestedSolve(parameters))
+    _nested_solve(NestedSolve(parameters)),
+
+    _prop_dFidci(2),
+    _dFidci_names(2)
 
 {
+  _prop_dFidci[0] = &declareProperty<Real>(_dFidci_names[0]);
+  _dFidci_names[0] = DerivativeMaterialPropertyNameInterface::derivativePropertyNameFirst(
+      _Fi_names[0], _ci_names[0]);
+  // _dFidci[0] = &getMaterialPropertyDerivative<Real>(_Fi_names[0], _ci_names[0]);
+
+  // declare _fi_copy to be passed in to the kernels
+  for (unsigned int m = 0; m < 2; ++m)
+  {
+    _prop_fi[m] = &getMaterialPropertyByName<Real>(_Fi_names[m]);
+    _fi_copy[m] = &declareProperty<Real>(_fi_copy_name[m]);
+  }
```

Comment:
@dschwen Is this how we declare the free energy copies? Line 179 assigns the free energy material to the copies.

Reply:
yeah. maybe just do `_fi_copy[m] = &declareProperty<Real>(_prefix + _Fi_names[m]);` where `_prefix` is a fixed string.